### PR TITLE
back-end APIs for dataset explore

### PR DESCRIPTION
**Pull request by cdathuraliya**

### CODE REVIEW

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java [hunk 1]
```diff
@@ -351,4 +351,20 @@ public void deleteDatasetVersion(int tenantId, String userName, long versionsetI
         }
     }
 
+    public List<String> getFeatureNames(long datasetId, String featureType) throws MLDataProcessingException {
+        try {
+            return databaseService.getFeatureNames(datasetId, featureType);
+        } catch (DatabaseHandlerException e) {
+            throw new MLDataProcessingException(e.getMessage(), e);
```
Comment (SupunS) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java:358]: Add a custom message to the exception

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java [hunk 2]
```diff
@@ -351,4 +351,20 @@ public void deleteDatasetVersion(int tenantId, String userName, long versionsetI
...
+        }
+    }
+
+    public String getSummaryStats(long datasetId, String featureName) throws MLDataProcessingException {
+        try {
+            return databaseService.getSummaryStats(datasetId, featureName);
+        } catch (DatabaseHandlerException e) {
+            throw new MLDataProcessingException(e.getMessage(), e);
```
Comment (SupunS) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java:366]: Add a custom message to the exception

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java (hunk 1 above)
Comment (cdathuraliya) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java:358]: I think here we are wrapping the message of DatabaseHandlerException from previous layer. Please check other methods.

File: components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java (hunk 2 above)
Comment (cdathuraliya) [inline components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java:366]: I think here we are wrapping the message of DatabaseHandlerException from previous layer. Please check other methods.

**Pull request merged**